File: examples/governance_static_go/domain/activated_agreement.go
package domain

import "time"

type ActivatedAgreement struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	AgreementId string `json:"agreement_id"`
	EffectiveDate time.Time `json:"effective_date"`
	ExpirationDate time.Time `json:"expiration_date"`
	ModelId string `json:"model_id"`
	DataSource string `json:"data_source"`
	Purpose string `json:"purpose"`
	ConsentType string `json:"consent_type"`
	Restrictions []Restriction `json:"restrictions"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e ActivatedAgreement) EventName() string { return "ActivatedAgreement" }

func (e ActivatedAgreement) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/model_registry_domain/lib/model_registry_domain/data_usage_agreement/events/activated_agreement.rb
module ModelRegistryDomain
  class DataUsageAgreement
    module Events
      class ActivatedAgreement
        attr_reader :aggregate_id, :agreement_id, :effective_date, :expiration_date, :model_id, :data_source, :purpose, :consent_type, :restrictions, :status, :occurred_at

        def initialize(aggregate_id: nil, agreement_id: nil, effective_date: nil, expiration_date: nil, model_id: nil, data_source: nil, purpose: nil, consent_type: nil, restrictions: nil, status: nil)
          @aggregate_id = aggregate_id
          @agreement_id = agreement_id
          @effective_date = effective_date
          @expiration_date = expiration_date
          @model_id = model_id
          @data_source = data_source
          @purpose = purpose
          @consent_type = consent_type
          @restrictions = restrictions
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
